### PR TITLE
[WIP] Try _final and _destroy

### PR DESCRIPTION
**Pull request by ralphtheninja**
This doesn't fix race condition so not sure how I'm supposed to use `_final`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "tape": "^4.9.0"
   },
   "scripts": {
-    "test": "standard && node test.js"
+    "test": "node test.js"
```
Comment (ralphtheninja) [inline package.json]: Temporary change.

File: level-ws.js [hunk 2]
```diff
@@ -129,10 +125,14 @@ WriteStream.prototype.toString = function () {
   return 'LevelUP.WriteStream'
 }
 
-WriteStream.prototype.destroy = function () {
-  if (this._destroyed) return
+WriteStream.prototype._final = function (cb) {
+  this._buffer = null
+  this.writable = false
+  process.nextTick(cb)
```
Comment (ralphtheninja) [inline level-ws.js]: Tried calling `cb()` directly, which didn't work, so changed to `process.nextTick` but also doesn't help.

Comment (vweevers) [reply level-ws.js]: The key here, I think, is to `db.batch` whatever is left in the buffer, then call the callback.

Comment (ralphtheninja) [reply level-ws.js]: So basically, call `_flush`.

File: level-ws.js [hunk 3]
```diff
@@ -129,10 +125,14 @@ WriteStream.prototype.toString = function () {
...
+}
+
+WriteStream.prototype._destroy = function () {
   this._buffer = null
-  this._destroyed = true
   this.writable = false
   this.emit('close')
```
Comment (vweevers) [inline level-ws.js]: I think this will emit `close` before `finish`.

**Comment by ralphtheninja** [comment]
@vweevers Currently I'm shooting from the hip and I don't _really_ know what's happening. I'll back up a bit and use current `master` to try and find out _why_ the race condition happens.

**Comment by vweevers** [comment]
@ralphtheninja I think two things are important:

1. Testing event order (similar to `iterator-stream`)
2. Adding state to the `process.nextTick` in `_write`:

https://github.com/Level/level-ws/blob/4fe02570ff07ebb0e3535cf45caeb5bc7e7e471d/level-ws.js#L84

Nothing waits for this function to complete.

**Pull request closed without merging**